src/public/components/restricted-input.js

```diff
@@ -0,0 +1,43 @@
+import K, * as U from "karet.util"
+import React     from "karet"
+
+export const RestrictedInput = ({value, meta: {format, parse}, ...props}) => {
+  const edited = U.atom(undefined)
```

Comment:
This comes from the examples repo, but nowadays I generally recommend exposing (by default) local state like this as parameters.